File: xanathar/Day01.cs
using System;
using System.Text;
using System.Text.RegularExpressions;
using System.Linq;
using System.Collections.Generic;
using System.IO;
using Helpers;

namespace Day01 {
    public class Test1 : DayBase
    {
        public override void Run() {
            int count = 0;
            var data = File.ReadAllLines("data/day1.txt").Where(s => !string.IsNullOrEmpty(s)).Select(s => int.Parse(s)).ToArray();

            for (int i = 0; i < data.Length; i++)
                if (i != 0 && data[i] > data[i - 1]) count++;

            Console.WriteLine($"{count}");
        }
    }

    public class Test2 : DayBase
    {
        public override void Run() {
            int count = 0;
            var data = File.ReadAllLines("data/day1.txt").Where(s => !string.IsNullOrEmpty(s)).Select(s => int.Parse(s)).ToArray();

            for (int i = 0; i < data.Length; i++)
                if (i >= 3 && data[i] > data[i - 3]) count++;

            Console.WriteLine($"{count}");
        }
    }
}

File: xanathar/rs/src/day01.rs
#![allow(unused_imports)]
use std::iter::Iterator;
use std::collections::{HashMap, HashSet, VecDeque, BTreeSet, BTreeMap};
use regex::Regex;
use lazy_static::lazy_static;
use std::ops::RangeInclusive;
use crate::utils;

pub fn test1() {
    let data = utils::parse_lines("../data/day1.txt", |s| s.parse::<u32>().ok());
    let mut count = 0u32;

    for i in 1..data.len() {
        if data[i] > data[i - 1] {
            count += 1;
        }
    }

    println!("{}", count);
}

pub fn test2() {
    let data = utils::parse_lines("../data/day1.txt", |s| s.parse::<u32>().ok());
    let mut count = 0u32;

    for i in 3..data.len() {
        if data[i] > data[i - 3] {
            count += 1;
        }
    }

    println!("{}", count);
}

